src/app/services/menu/menu.service.ts

```diff
@@ -182,48 +220,53 @@ export class MenuService extends withUnsubscribe() {
   }
 
   private getSecondaryMenuData(page: IPageInfo): SecondaryMenuData {
-    // get current page
-    const current = page.pageRoute;
-    current.active = true; // Ignore predicate
-
-    // and parent pages
-    const parentMenuRoutes: MenuRoute[] = [];
-    let menuRoute: MenuRoute = current;
-    while (menuRoute.parent) {
-      menuRoute = menuRoute.parent;
-      menuRoute.active = true; // Ignore predicate
-      parentMenuRoutes.push(menuRoute);
-    }
+    // Get current page and all parents
+    const parentMenuRoutes = this.rootToMenuRoute(page.pageRoute).map(
+      (menuRoute) => menuRoute
+    );
 
-    // and add it all together
     const widgets = page.menus?.linkWidgets?.toOrderedSet() ?? OrderedSet();
+
+    // Combine current route ancestry with default menu
     const links = this.defaultMenu.contextLinks
-      .concat<NavigableMenuItem | MenuModalWithoutAction>(
-        page.menus?.links ?? OrderedSet(),
-        OrderedSet(parentMenuRoutes).reverse(), // List lineage correctly
-        OrderedSet([current])
-      )
-      .filter((link) => this.filterByPredicate(link, page))
+      .concat(page.menus?.links ?? OrderedSet<NavigableMenuItem>())
+      .filter((link): boolean => this.filterByPredicate(link, page))
+      // Add parent menu routes after predicate so they are not filtered out
+      .concat(parentMenuRoutes)
       .sort(this.sortByOrderAndIndentation);
 
-    // Save to state
     return { links, widgets };
   }
 
+  private rootToMenuRoute(menuRoute: MenuRoute): OrderedSet<MenuRoute> {
+    let menuRoutes = OrderedSet<MenuRoute>([menuRoute]);
+    let current: MenuRoute = menuRoute;
+    while (current.parent) {
+      current = current.parent;
+
+      if (
+        this.config.settings.hideProjects &&
```

Comment:
you might pull this value out of the loop and assign it to a const
```suggestion
        hideProjects &&
```